include/PolygonSimplification.hpp

```diff
@@ -33,62 +33,107 @@
 
 namespace com {
 	namespace geopipe {
-		namespace PolySimp {
-			using CGALKernel = CGAL::Exact_predicates_exact_constructions_kernel;
-			using CGALPoint = CGALKernel::Point_2;
-			using CGALPolygon = CGAL::Polygon_2<CGALKernel>;
+		template<typename Kernel = CGAL::Exact_predicates_exact_constructions_kernel, typename Point = typename Kernel::Point_2>
+		class PolySimp {
+			using CGALKernel = Kernel;
+			using CGALPoint = Point;
+			using CGALPolygon = CGAL::Polygon_2<CGALKernel, std::vector<Point>>;
```

Comment:
A couple questions:

(a) it doesn't seem like `Point` should be able to be specified separately from `Kernel`?
(b) are there any restrictions on the types of kernels that can be used? I would be fairly surprised if an inexact kernel could be used with Nef polyhedra....

Reply:
(a) In an ideal world only `Point` would be specified and we would derive the `Kernel` from the point, but my reading of https://doc.cgal.org/latest/Kernel_23/classCGAL_1_1Point__2.html doesn't show a way to do this.
(b) I'd be surprised as well. The purpose of this PR is really to allow different point types; the different kernels is just a side-effect based on (a).

Reply:
Why are you trying to use a point type other than `Kernel::Point_2`?

Reply:
But if you really want to just specify `Point`, you need something like:

```c++
namespace detail {
    template<class Point>
    struct cgal_infer_point_kernel { using type = void; }

    template<class Kernel>
    struct cgal_infer_point_kernel< Point_2<Kernel> > { using type = Kernel; }

}

template<class Point, class Kernel = typename detail::cgal_infer_point_kernel<Point>::type >
/* ... */
```

Reply:
ah, okay, so, based on our Google Chat discussion, there's no reason to specify point type and kernel type separately. Following up with additional change request.